sm/pmp.c

```diff
@@ -238,6 +233,25 @@ static void send_and_sync_pmp_ipi(int region_idx, enum ipi_type type, uint8_t pe
   }
 }
 
+void pmp_ipi_update() {
+  if (ipi_mailbox[read_csr(mhartid)].pending) {
+    if(ipi_type == IPI_PMP_SET) {
+      uint8_t perm = ipi_mailbox[read_csr(mhartid)].perm;
+      pmp_set(ipi_region_idx, perm);
+    } else {
+      pmp_unset(ipi_region_idx);
+    }
+
+    ipi_mailbox[read_csr(mhartid)].pending = 0;
+  }
+}
+
+void pmp_ipi_acquire_lock() {
+  while(spinlock_trylock(&pmp_ipi_global_lock)) {
```

Comment:
Please add comment

Reply:
I have just added comments to this function and the other functions which I made.